.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI
+
+on: [push, pull_request]
+
+permissions: "read-all"
+
+jobs:
+  lint:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: pip
+          cache-dependency-path: pyproject.toml
+      - name: "Install dependencies"
+        run: python -m pip install .[develop]
+      - uses: pre-commit/action@v3.0.0
```

Comment:
As I am not super familiar with GitHub actions, I have one general question; seems that we are referencing various versioned "plugins" like `pre-commit/action@v3.0.0` etc.

I assume that they get regular updates, maybe even have security issues, breaking with newer Python versions etc.

What is the workflow for keeping up to date with them? Should we just expect that when there is a problem the PR check will fail, or is there something more automated/proactive that can be used?

Reply:
There are two scenarios when it's useful to upgrade:

 * When there's a security issue (but we're not doing anything sensitive and there's no credentials to steal and the actions has read-only permissions)
 * When the version we use is deprecated (it shows up in the summary of the build, so we'll end up seeing it, https://github.com/elastic/rally-tracks/pull/346 is a nice example here)

We can use dependabot (which is now owned by GitHub) and configure it to open pull requests in case of security issues. Otherwise there's not much value in upgrading. Upgrading for every change is going to add a lot of noise.

I'll look into configuring dependabot.

Reply:
I enabled Dependabot security updates in the settings section of this repository:

![image](https://user-images.githubusercontent.com/42327/219314962-1e9d6cd5-a539-4b6f-95b3-cc4bfde01604.png)

Let's see if the defaults work for us. If not, we can configure them using a dependabot.yml file: https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/configuring-dependabot-security-updates